### ISSUE TITLE
[Native app] Crashing/Freezing on Kubuntu Linux 18.04 

### ISSUE THREAD

**Issue by Alexand3r** [a85/POSTMan-Chrome-Extension#6297]
**Describe the bug**
Postman app crashes completely or goes black screen/unresponsive UI on opening/editing/saving requests. There is not a "sure" path to hit this bug, it happens like it is randomly, the most ways I got it to crash is by creating new unsaved requests (around 5-7).

**App information (please complete the following information):**
 - App Type: Native Type
 - Postman Version: 7.0.7(snap) / 7.0.8 / 7.0.9
 - OS: Kubuntu 18.04 x64

**Additional context**
Looking in the terminal it looks like an Out of memory issue? However my laptop has 12GB RAM
Shell log 1
```
6:26:45 - andrei[~]$ postman 
1555338409210 main info "Booting Postman 7.0.8, linux-4.15.0-47-generic on x64"
1555338409212 main info "EventBus~initialize - Success"
1555338409224 main info "UpdateHandler~init - Success"
1555338409227 main info "RuntimeExecutionService~initialized: Success"
1555338409229 main info "LinuxAutoUpdater~Cleanup - Initial cleanup successful"
1555338409289 main info "ProtocolHandler~init - Success with status: false]"
1555338409301 main info "PluginHost~bootHost: Booted - 1"
1555338409466 main info "Bootstrap-models~bootstrap - Success"
1555338409750 pluginHost info "Booting Plugin Host for Postman 7.0.8, linux-4.15.0-47-generic on x64"
1555338409755 pluginHost info "PluginsHost: Booted"
1555338409763 pluginHost info "PluginInstallationService~_shouldReInstallPlugin",{"installedVersion":"0.0.10","currentVersion":"0.0.10"}
1555338409764 pluginHost info "PluginInstallationService~_shouldReInstallPlugin",{"installedVersion":"0.0.1","currentVersion":"0.0.1"}
1555338409764 pluginHost info "PluginsHost~installOrUpdatePlugins: Synced all plugins",[{"pluginId":"postman.openApi","version":"0.0.10","status":"skipped"},{"pluginId":"postman.sample","version":"0.0.1","status":"skipped"}]
1555338411572 main info "Main~AppEvents - Received booted event for process shared"
1555338414039 main info "UpdateHandler~app-update-events - Received event",{"name":"checkForElectronVersionUpdated","namespace":"appUpdate"}
1555338414044 main info "Main~AppEvents - Received booted event for process requester"

#
# Fatal process OOM in heap setup
#

1555338435459 main info "WindowManager~closeHandler - Closed Window (id: 3 )"
1555338435512 main info "Quitting app"
1555338435519 main info null
1555338435578 main error {"name":"Error","message":"PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT","stack":"Error: PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT\n    at ChildProcess.host.on (/home/andrei/Desktop/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:63:25)\n    at emitTwo (events.js:125:13)\n    at ChildProcess.emit (events.js:213:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)"}
1555338435582 main error {"name":"Error","message":"PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2","stack":"Error: PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2\n    at PluginHost.bootHost (/home/andrei/Desktop/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:26:9)\n    at <anonymous>\n    at process._tickDomainCallback (internal/process/next_tick.js:228:7)"}
```
Shell log 2
```
16:49:21 - andrei[~]$ postman 
1555339921364 main info "Booting Postman 7.0.8, linux-4.15.0-47-generic on x64"
1555339921366 main info "EventBus~initialize - Success"
1555339921379 main info "UpdateHandler~init - Success"
1555339921382 main info "RuntimeExecutionService~initialized: Success"
1555339921384 main info "LinuxAutoUpdater~Cleanup - Initial cleanup successful"
1555339921433 main info "ProtocolHandler~init - Success with status: false]"
1555339921444 main info "PluginHost~bootHost: Booted - 1"
1555339921596 main info "Bootstrap-models~bootstrap - Success"
1555339921885 pluginHost info "Booting Plugin Host for Postman 7.0.8, linux-4.15.0-47-generic on x64"
1555339921889 pluginHost info "PluginsHost: Booted"
1555339921896 pluginHost info "PluginInstallationService~_shouldReInstallPlugin",{"installedVersion":"0.0.10","currentVersion":"0.0.10"}
1555339921896 pluginHost info "PluginInstallationService~_shouldReInstallPlugin",{"installedVersion":"0.0.1","currentVersion":"0.0.1"}
1555339921896 pluginHost info "PluginsHost~installOrUpdatePlugins: Synced all plugins",[{"pluginId":"postman.openApi","version":"0.0.10","status":"skipped"},{"pluginId":"postman.sample","version":"0.0.1","status":"skipped"}]
1555339923665 main info "Main~AppEvents - Received booted event for process shared"
1555339926057 main info "UpdateHandler~app-update-events - Received event",{"name":"checkForElectronVersionUpdated","namespace":"appUpdate"}
1555339926059 main info "Main~AppEvents - Received booted event for process requester"

#
# Fatal process OOM in heap setup
#

1555339950058 main info "WindowManager~closeHandler - Closed Window (id: 3 )"
1555339950100 main info "Quitting app"
1555339950102 main info null
1555339950112 main error {"name":"Error","message":"PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT","stack":"Error: PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT\n    at ChildProcess.host.on (/home/andrei/Desktop/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:63:25)\n    at emitTwo (events.js:125:13)\n    at ChildProcess.emit (events.js:213:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)"}
1555339950116 main error {"name":"Error","message":"PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2","stack":"Error: PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2\n    at PluginHost.bootHost (/home/andrei/Desktop/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:26:9)\n    at <anonymous>\n    at process._tickDomainCallback (internal/process/next_tick.js:228:7)"}
```

![Screenshot_20190415_165422](https://user-images.githubusercontent.com/988056/56142553-29f6f500-5f9f-11e9-88b6-23512e57a377.png)

**Comment by numaanashraf** [comment]
@Alexand3r Thanks for reporting this. Looks like Postman ran out of memory. Could you share your system specs including memory and CPU?

**Comment by prakhar-pal** [comment]
Same issue is happening with me. Only 39% of memory is used at 6.9 GB of 18GB

Specs are
OS: elementary OS(based on Ubuntu 18.04)
RAM: 18 GB
CPU: Quad-Core Intel® Core™ i7-4770 CPU @ 3.40GHz

**Comment by Alexand3r** [comment]
@numaanashraf Sorry for the long wait.

EDIT: I can also confirm the bug is still present in 7.0.9

Intel Core i3 7100U @ 2.4GHz 4 cores
12GB RAM

![Screenshot_20190503_112055](https://user-images.githubusercontent.com/988056/57129043-96daff00-6d95-11e9-886f-fc00c655024a.png)

**Comment by soymanuvalle** [comment]
Same happening over here...

OS: elementary OS (based on Ubuntu 18.04)
RAM: 12 GB
CPU: Intel® Core™ i7 7700HQ Processor 3.80 GHz

```
/usr/local/Postman$ sudo ./Postman
1557430280274 main info "Booting Postman 7.0.9, linux-4.15.0-48-generic on x64"
1557430280275 main info "EventBus~initialize - Success"
1557430280288 main info "UpdateHandler~init - Success"
1557430280290 main info "RuntimeExecutionService~initialized: Success"
1557430280291 main info "LinuxAutoUpdater~Cleanup - Initial cleanup successful"
1557430280318 main info "ProtocolHandler~init - Success with status: false]"
1557430280327 main info "PluginHost~bootHost: Booted - 1"
1557430280518 main info "Bootstrap-models~bootstrap - Success"
1557430280633 pluginHost info "Booting Plugin Host for Postman 7.0.9, linux-4.15.0-48-generic on x64"
1557430280635 pluginHost info "PluginsHost: Booted"
1557430280639 pluginHost info "PluginInstallationService~_shouldReInstallPlugin",{"installedVersion":"0.0.10","currentVersion":"0.0.10"}
1557430280640 pluginHost info "PluginInstallationService~_shouldReInstallPlugin",{"installedVersion":"0.0.1","currentVersion":"0.0.1"}
1557430280640 pluginHost info "PluginsHost~installOrUpdatePlugins: Synced all plugins",[{"pluginId":"postman.openApi","version":"0.0.10","status":"skipped"},{"pluginId":"postman.sample","version":"0.0.1","status":"skipped"}]
1557430281916 main info "Main~AppEvents - Received booted event for process shared"
1557430283462 main info "UpdateHandler~app-update-events - Received event",{"name":"checkForElectronVersionUpdated","namespace":"appUpdate"}
1557430283464 main info "Main~AppEvents - Received booted event for process requester"
^C1557430332912 main info "Quitting app"
1557430332913 main info null
1557430332917 main error {"name":"Error","message":"PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT","stack":"Error: PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT\n    at ChildProcess.host.on (/usr/local/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:63:25)\n    at emitTwo (events.js:125:13)\n    at ChildProcess.emit (events.js:213:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)"}
1557430332919 main error {"name":"Error","message":"PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2","stack":"Error: PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2\n    at PluginHost.bootHost (/usr/local/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:26:9)\n    at <anonymous>\n    at process._tickDomainCallback (internal/process/next_tick.js:228:7)"}
```

**EDIT:** for some reason it has been fixed after uninstalling and installing again with snap instead of the website download.

**Comment by asule90** [comment]
having the same problem on ubuntu 18, after updated to postman 7.0.9.

getting this log:
```
<memory>:1: Invalid color constant '##d3d3d3'
Gtk-Message: 16:21:07.768: (for origin information, set GTK_DEBUG): failed to retrieve property `ChromeGtkFrame::frame-color' of type `GdkColor' from rc file value ""##d3d3d3"" of type `gchararray'
1557739267888 main info "Booting Postman 7.0.9, linux-4.15.0-39-generic on x64"
1557739267889 main info "EventBus~initialize - Success"
1557739267901 main info "UpdateHandler~init - Success"
1557739267903 main info "RuntimeExecutionService~initialized: Success"
1557739267905 main info "LinuxAutoUpdater~Cleanup - Initial cleanup successful"
1557739268166 main info "ProtocolHandler~init - Success with status: true]"
1557739268179 main info "PluginHost~bootHost: Booted - 1"
1557739268224 main info "Quitting app"
1557739268232 main error "WindowManager~getSharedWindow: Shared window is not available"
1557739268233 main info null
1557739268241 main error {"name":"Error","message":"PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT","stack":"Error: PluginHost~hostErrorOrExit: Exited with code null with signal SIGINT\n    at ChildProcess.host.on (/home/asule/Apps/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:63:25)\n    at emitTwo (events.js:125:13)\n    at ChildProcess.emit (events.js:213:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)"}
1557739268242 main error {"name":"Error","message":"PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2","stack":"Error: PluginHost~bootHost: Disabling boot since it has exceeded the maximum allowed boot count 2\n    at PluginHost.bootHost (/home/asule/Apps/Postman/app/resources/app/node_modules/@postman/app-plugins-host/index.js:26:9)\n    at <anonymous>\n    at process._tickDomainCallback (internal/process/next_tick.js:228:7)"}

```

read [this](https://github.com/postmanlabs/postman-app-support/issues/4374#issuecomment-439020387) and [this](https://github.com/postmanlabs/postman-app-support/issues/4359#issuecomment-374623031) and works like a charm :+1: :confetti_ball:

**Comment by Alexand3r** [comment]
Tried what @asule90 suggested, unfortunately I have the same issue

EDIT: I've tried with snap too, same story, deleted everything and cleaned...

![Screenshot_20190515_144518](https://user-images.githubusercontent.com/988056/57776571-241c3d00-7720-11e9-810b-d8d684c9b2f4.png)

**Comment by Alexand3r** [comment]
I've figured it out, I think.... 

TL;DR: Set `ulimit -v unlimited`

It wasn't completely related to Postman, rather more tied to Kubuntu / Ubuntu's `ulimit -v` which was default set(?) to `625000`,  I've edited my `.bashrc` to have set `ulimit -v unlimited` and I can see that Postman stopped crashing, I'll close the issue in the next days if this fixed it for me

Edit:  After 2hrs, Postman starved the system of memory and procs started crashing one by one....

**Comment by gervaisb** [comment]
Any news on that ? Same issue with Ubuntu 19.04.

This is blocking us to use Postman for testing our APIs.

**Comment by DannyDainton** [comment]
Hey @gervaisb 

Is this still an issue for you with the latest version?

**Comment by mccannt** [comment]
Closing out this issue for now. 
If anyone is still experiencing this issue, please let us know.
